pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -106,6 +132,72 @@ func (im *Image) build(sourceResolver *SourceResolver, builder AbstractBuilder,
 	}
 }
 
+func (im *Image) rebase(builder AbstractBuilder, previousRunImage string, nextBuildNumber int64) *Build {
+	buildNumber := strconv.Itoa(int(nextBuildNumber))
+	return &Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    im.Namespace,
+			GenerateName: im.generateBuildName(buildNumber),
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(im),
+			},
+			Labels: im.labels(map[string]string{
+				BuildNumberLabel: buildNumber,
+				ImageLabel:       im.Name,
+			}),
+			Annotations: map[string]string{
+				BuildReasonAnnotation: BuildReasonStack,
+			},
+		},
+		Spec: BuildSpec{
```

Comment:
Is it problematic for the build needed logic if the last build does not have all the current source spec like last git url etc?

Reply:
Oh yeah..we kinda just assumed everything is grabbed from the status and not the spec. We would need to put that information back. Though it "feels" weird for the rebase spec, I think it needs to be there so that the next build has it available to it.

Reply:
Yeah, that's why I wonder if the build is the one that should `decide` to rebase instead of the image.